### ISSUE TITLE
Feature: link to Callerlab definitions from Dance Programs

### ISSUE THREAD

**Issue by mpogue2** [mpogue2/SquareDesk#167]
This would be nice for callers who are teaching.  Suggested by R Branson.

I suppose it could be a link, or maybe hover text.  Could be automatic, or specified in the dance program list.  A number of possibilities here!

**Comment by mhkull** [comment]
I like the sqview ideo of offering tabs that can be "filled" with caller specific content.

**Comment by mpogue2** [comment]
@mhkull Do you have a picture of what this looks like?  I've never used the sqview feature.

**Comment by mhkull** [comment]
[http://caller.martinkull.de/howIusesqview.html](http://caller.martinkull.de/howIusesqview.html)

**Comment by mpogue2** [comment]
In the mean time, we could link call names to Taminations: http://www.tamtwirlers.org/tamination/ms/hinge.html or, we could link to Vic Ceder: (but Vic doesn't have links for everything, e.g. the MS version of Hinge!).

We could add a column to the dance programs input CSV file that contains the URL to link to.  The visible column in the app that shows the call names would be the same width as today, but all of the calls would be blue-underlined, indicating that you can click on them.  (or, should we do the "only show blue highlighting when hovering over it" thing?  That might be cleaner.)

Should we just open up the link in the default (external) browser (that would be easiest, and allow multiple tabs, security, etc.)?  

Or, should we put in a Reference tab that is a browser, and when the user clicks on a call we switch to it and show the URL (this almost provides full Reference tab capability, by just making the initial index.html the one in the reference folder, but it's a very limited browser)?

**Comment by mpogue2** [comment]
I am leaning toward linking to an external browser using the Taminations link, which provides a very nice animation, as well as the Callerlab definition, for each of the calls.

**Comment by danlyke** [comment]
So we can now fill the References tab with whatever we'd like. I kinda like the idea of hover popups.

I'm not so keen on external links, because I don't have net connectivity when I'm calling (and I may switch to my new snazzy laptop, but my calling laptop doesn't have enough RAM to comfortably run Firefox, let alone alongside SquareDesk).

**Comment by mpogue2** [comment]
I agree...I explicitly turn off Wi-Fi when calling, so I get no notifications or popups or dings or anything else while calling.  Heaven help the Windows users who leave Wi-Fi on while calling -- more than once I've seen a Windows laptop decide that it needed to upgrade the OS, DURING THE DANCE, and BOOM the whole dance was stopped dead, courtesy of Microsoft.
